### PR TITLE
Fixing slow search queries on `bugfix/casing-json-fields-#94` branch #658

### LINKED ISSUE #658: Fix investigationSearchTable slow queries due to how facility cycles are fetched
**Description**:
Investigating failing e2e tests with Matt on his JSON casing branch, we discovered that the inclusion of `facilityCycle` in the query is massively slowing down the request. Including `facilityCycles` means a lot of duplicate data since it fetches the whole list of facility cycles for each investigation. Instead, we should send a separate request to the API for the list of facility cycles (if we detect we're an ISIS facility), store this in the table as local state and use that for the link construction instead.

**Acceptance criteria**:
- [ ] Facility Cycles are fetched once per investigationSearchTable and not for each item in a table

### PR DESCRIPTION
**Pull request by sam-glendenning**
## Description
This issues a fix for slow queries in the investigationSearchTable that are being experienced on the `bugfix/casing-json-fields-#94` branch. As the linked issue describes, what was making the queries slow was that the entire list of facility cycles was being fetched with each query to then construct the link to each investigation. This resulted in a large amount of data being sent that wasn't necessary.

This fix fetches the entire list of facility cycles on component mount, stores that list in local state and uses that to construct the investigation links instead (ISIS investigations only). Facility cycles are no longer fetched in the investigation queries, which still take place.

## Testing instructions
- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking
Connect to Issue #94 and [PR #663](https://github.com/ral-facilities/datagateway/pull/663)
Closes #658

### CODE REVIEW

**Comment by sam-glendenning** [comment]
As of 24/05/21, this is up to date with the branch it is based off of, [bugfix/casing-json-fields-#94](https://github.com/ral-facilities/datagateway/tree/bugfix/casing-json-fields-%2394). As issue #658 relates to this branch, I have made it the base to be merged into. Alternatively, as there's not much code to it, I can simply commit it directly to the base branch instead, although I thought this PR would be good for review purposes.

**Comment by MRichards99** [comment]
Fixed the Search e2e tests so they now pass, this change now means some of the timeouts can be reduced too :)

image

**Review by MRichards99 — APPROVED** [review]
This fixes the issue mentioned in the original issue and the Search e2e tests now pass. I think @louise-davies mentioned some visit ID changes got reverted and it might be good for one of you to sanity check my e2e test changes, but I'm happy for this to get merged once those things happen :)

**Comment by sam-glendenning** [comment]
Will wait on further info from Louise about changes I may have overwritten before proceeding.

File: packages/datagateway-search/src/table/investigationSearchTable.component.tsx [hunk 1]
```diff
@@ -229,7 +254,7 @@ const InvestigationSearchTable = (
           filterComponent: textFilter,
         },
         {
-          label: t('investigations.visit_id'),
+          label: t('investigations.visitId'),
```
Comment (louise-davies) [inline packages/datagateway-search/src/table/investigationSearchTable.component.tsx:257]: Can this change be reverted please - this undoes the fix to #696

**Pull request merged**